### PR TITLE
docs: Update guides to production with SASL/TLS

### LINKED ISSUE #612: Create new quickstart guide for production
# This is a placeholder.

I will come back and fill in more details. This is a part of #592

Mostly a copy of the developer quickstart guide, but enabling tls and sasl, and disabling developer mode. Reference external access page (yet to be written). The default expectation is that the user will provide the client application within the same kubernetes cluster as redpanda.

Add a section to talk about performance expectations with specific architecture.

No "clean-up" section.

---

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: February 17

Fixes #612

- Updated the managed Kubernetes guides with production settings (SASL and TLS)
- Backported changes to 22.3
- Added some troubleshooting tips for SASL and TLS
- Added `--reuse-values` flag to `helm upgrade` examples
- Provided tips on configuring Console
- Added troubleshooting steps for errors during `helm upgrade`

### CODE REVIEW

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx [hunk 1]
```diff
@@ -13,27 +13,26 @@ import Versions from '../../../../shared/_versions.mdx'
...
+Deploy a local Redpanda cluster with Redpanda Console using the Helm chart. Explore the essentials of how Redpanda works in Kubernetes and what components are deployed by default. Then, use rpk both as an internal client and an external client to interact with your Redpanda cluster from the command line.
 
 <OperatorNote/>
 
 ## Prerequisites
 
 Before you begin, make sure that you have the correct software for your Kubernetes platform:
 
-<Tabs groupId="kubernetes-platforms">
+<Tabs groupId="kubernetes-platform" queryString>
```
Comment (shorgi) [inline docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:29]: The tab grouping no longer works: choosing 'kind' or 'minikube' here no longer updates the tabs below, as it did on https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/kubernetes/local-guide/

Comment (JakeSCahill) [reply docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:29]: good spot 😄 I'll look into that, seems weird

**Review by RafalKorepta — APPROVED** [review]

File: docs/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_external-access-steps.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import InstallRpkHomebrew from "../../../../../../get-started/shared/_install-rpk-homebrew.mdx"
+import InstallRpkLinux from "../../../../../../get-started/shared/_install-rpk-linux.mdx"
+
+1. Add mappings in your `/etc/hosts` file between your worker nodes' IP addresses and their custom domain names:
+
+  ```bash
+  kubectl -n redpanda get endpoints,node -A -o go-template='{{ range $_ := .items }}{{ if and (eq .kind "Endpoints") (eq .metadata.name "redpanda-external") }}{{ range $_ := (index .subsets 0).addresses }}{{ $nodeName := .nodeName }}{{ $podName := .targetRef.name }}{{ range $node := $.items }}{{ if and (eq .kind "Node") (eq .metadata.name $nodeName) }}{{ range $_ := .status.addresses }}{{ if eq .type "ExternalIP" }}{{ .address }} {{ $podName }}.${DOMAIN}{{ "\n" }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}' | envsubst | sudo tee -a /etc/hosts
```
Comment (alejandroEsc) [inline docs/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_external-access-steps.mdx:9]:
```suggestion
sudo true && kubectl -n redpanda get endpoints,node -A -o go-template='{{ range $_ := .items }}{{ if and (eq .kind "Endpoints") (eq .metadata.name "redpanda-external") }}{{ range $_ := (index .subsets 0).addresses }}{{ $nodeName := .nodeName }}{{ $podName := .targetRef.name }}{{ range $node := $.items }}{{ if and (eq .kind "Node") (eq .metadata.name $nodeName) }}{{ range $_ := .status.addresses }}{{ if eq .type "ExternalIP" }}{{ .address }} {{ $podName }}.${DOMAIN}{{ "\n" }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}{{ end }}' | envsubst | sudo tee -a /etc/hosts
```

**Review by alejandroEsc — APPROVED** [review]
One item to take care of osx, thank you!

File: docs/deploy/deployment-option/self-hosted/kubernetes/aks-guide.mdx [hunk 3]
```diff
@@ -79,38 +88,6 @@ In this step, you create three worker nodes: one worker node for each Redpanda b
   az aks get-credentials --resource-group redpandaResourceGroup --name redpanda
   ```
 
-## Deploy Redpanda
-
-<DeployRedpanda/>
-
-## Deploy Redpanda Console
-
-<DeployConsole/>
```
Comment (shorgi) [inline docs/deploy/deployment-option/self-hosted/kubernetes/aks-guide.mdx:88]: Is there an absolute link to RP Console's docs/README we can/should provide in the partial?

**Review by shorgi — APPROVED** [review]
Nice work! See minor suggestions.

File: docs/manage/kubernetes/troubleshooting/troubleshoot.mdx [hunk 4]
```diff
@@ -144,3 +144,20 @@ This error appears when a cluster has TLS enabled, and you try to access the adm
 If you're using rpk, make sure to include the `--tls-enabled`, `--admin-api-tls-truststore`, and `--api-urls` flags.
 
 For all available flags, see the [rpk command reference](../../../../reference/rpk).
+
+## UPGRADE FAILED: cannot patch "redpanda" with kind StatefulSet
+
+This error appears when you run `helm upgrade` without overriding all your previous value overrides.
```
Comment (joejulian) [inline docs/manage/kubernetes/troubleshooting/troubleshoot.mdx:150]:
```suggestion
This error appears when you run `helm upgrade` when using the `--values` flag but not including all of your overrides.
```

File: versioned_docs/version-22.3/manage/kubernetes/troubleshooting/troubleshoot.mdx [hunk 5]
```diff
@@ -138,3 +138,20 @@ This error appears when a cluster has TLS enabled, and you try to access the adm
 If you're using rpk, make sure to include the `--tls-enabled`, `--admin-api-tls-truststore`, and `--api-urls` flags.
 
 For all available flags, see the [rpk command reference](../../../../reference/rpk).
+
+## UPGRADE FAILED: cannot patch "redpanda" with kind StatefulSet
+
+This error appears when you run `helm upgrade` without overriding all your previous value overrides.
```
Comment (joejulian) [inline versioned_docs/version-22.3/manage/kubernetes/troubleshooting/troubleshoot.mdx:144]:
```suggestion
This error appears when you run `helm upgrade` when using the `--values` flag but not including all of your overrides.
```

`--set` is not affected

File: versioned_docs/version-22.3/manage/kubernetes/troubleshooting/troubleshoot.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+Error: parse error at (redpanda/templates/statefulset.yaml:203): function "dig" not defined
+```
+
+Make sure that you are using the minimum required version: <Versions name="helm"/>.
+
+```bash
+helm version
+```
+
+## repository name (redpanda) already exists
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply versioned_docs/version-22.3/manage/kubernetes/troubleshooting/troubleshoot.mdx:84]: Where there's an error, I think it's helpful to include the important part as a heading to make it searchable. I'll comb through them and add the full error to code blocks too.

File: docs/deploy/deployment-option/self-hosted/kubernetes/eks-guide.mdx [hunk 7]
```diff
@@ -131,169 +389,167 @@ Before you begin, you must have the following:
...
-  31092 is the Kafka API port that's exposed by the default NodePort Service.
-  :::
-
-1. Get the cluster info:
-
-  ```bash
-  rpk cluster info
-  ```
+:::note
+The `post-install` Job updates the Redpanda runtime configuration.
```
Comment (RafalKorepta) [inline docs/deploy/deployment-option/self-hosted/kubernetes/eks-guide.mdx:450]: Not it's called configuration

Comment (RafalKorepta) [reply docs/deploy/deployment-option/self-hosted/kubernetes/eks-guide.mdx:450]:
```
redpanda-0                          1/1     Running     0          17m
redpanda-1                          1/1     Running     0          17m
redpanda-2                          1/1     Running     0          17m
redpanda-configuration-smqv7        0/1     Completed   0          17m
redpanda-console-59f64b66d6-mmmgz   1/1     Running     0          17m
```

Comment (RafalKorepta) [reply docs/deploy/deployment-option/self-hosted/kubernetes/eks-guide.mdx:450]:
```
kubectl get job
NAME                     COMPLETIONS   DURATION   AGE
redpanda-configuration   1/1           5s         17m
```

Comment (JakeSCahill) [reply docs/deploy/deployment-option/self-hosted/kubernetes/eks-guide.mdx:450]: Ah, we renamed it 😅 thanks @RafalKorepta for the eagle eyes

**Review by joejulian — APPROVED** [review]

**Pull request merged**